### PR TITLE
Update SparseOps.h to add missing #endif

### LINKED ISSUE #14080: Missing #endif in sparseops
## Bug Description
After `./configure --with-derivative-type=sparse`, this file gets sourced and you can't compile due to missing endif.

## Steps to Reproduce
in framework,
```
./configure --with-derivative-type=sparse
make
```

## Impact
Remove need to have to patch manually

### PR DESCRIPTION
**Pull request by veeshy**
Closes #14080

### CODE REVIEW

File: framework/include/utils/SparseOps.h [hunk 1]
```diff
@@ -36,3 +36,4 @@ derivInsert(SemiDynamicSparseNumberArray<Real, unsigned int, NWrapper<N>> & deri
   derivs.insert(index) = value;
 }
 }
+#endif
```
Comment (andrsd) [inline framework/include/utils/SparseOps.h]: This should be moved 2 lines up.

**Review by andrsd — CHANGES_REQUESTED** [review]

**Comment by andrsd** [comment]
It is still wrong. You moved it just 1 line up. There are 2 `{` outside the `#ifdef`, so there needs to be 2 `}` after the `#endif`.

And,  please, squash the patches into a single commit, we really do not need 3 commits for a one line change...

**Comment by veeshy** [comment]
Third try is the charm ::facepalm::, also made it back into 1 commit

**Comment by andrsd** [comment]
BTW: you need to reference the issue number in your commit message, the precheck will fail if it is just in the PR description...

**Comment by veeshy** [comment]
Tagged the issue in the commit

**Review by andrsd — APPROVED** [review]

**Pull request merged**